### ISSUE TITLE
unresolved symbol class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > testing::FLAGS_gtest_filter

### ISSUE THREAD

**Issue by jmecosta** [google/googletest#2145]
hi,

after taking latest master, and compile gtest as shared library i get this:
1>libgeometryTests.obj : error LNK2001: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > testing::FLAGS_gtest_filter" (?FLAGS_gtest_filter@testing@@3V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@A)
1>D:\prod\kits\toolkits\BuildDrop\Test\v141\x64\Release\libgeometryTests.exe : fatal error LNK1120: 1 unresolved externals

this is our macro that was earlier working

`inline void SET_GTEST_FILTER(const std::string &InFilter)
{
    if(testing::GTEST_FLAG(filter) == "*")
    {
        ::testing::GTEST_FLAG(filter) = InFilter.c_str();
    }
}`

we have this beause gtest tries to run tests that are in other dlls. when building a shared lib

**Comment by gennadiycivil** [comment]
@jmecosta Could you please provide a bit more information, it is hard to follow. The best would be a small reproducible tester. 
Also please make sure that you are only using one copy? there may be another pre-compiled copy lurking of the system

**Comment by andry81** [comment]
I had the same issue with a static library linked versus gtest static library in the Visual Studio 2017 (cmake generated solution). In mine case the gtest was added in the cmake through the `add_subdirectory` function.
I've found that is not enough just add `GTEST_LINKED_AS_SHARED_LIBRARY=1` as a definition in the end target. You have to additionally add that to all static target on the way after the gtest static library:

```
myapp -> gtest.lib
myapp -> mylib.lib -> gtest.lib
```
Here additionally to the `myapp` target the `mylib` target must have `GTEST_LINKED_AS_SHARED_LIBRARY=1` definition too, even if `myapp` is already linked with the `gtest`. Otherwise the Visual Studio linker would ignore the first one linkage.
It works even if `myapp` does not have the `GTEST_LINKED_AS_SHARED_LIBRARY=1` definition.
